lang/typescript/src/utils/script.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+export type Script =
+  | 'Arabic'
+  | 'Han'
+  | 'Hangul'
+  | 'Hiragana'
+  | 'Katakana'
+  | 'Latin'
+  | 'Thai';
+
+export const scriptRegexes: Record<Script, RegExp> = {
+  Arabic: /\p{Script=Arabic}/u,
+  Han: /\p{Script=Han}/u,
+  Hangul: /\p{Script=Hangul}/u,
+  Hiragana: /\p{Script=Hiragana}/u,
+  Katakana: /\p{Script=Katakana}/u,
+  Latin: /\p{Script=Latin}/u,
+  Thai: /\p{Script=Thai}/u,
+};
```

Comment:
I took this list from https://github.com/Shopify/worldwide/blob/main/lib/worldwide/scripts.rb, but there's actually a bunch more Regex supports: https://www.regular-expressions.info/unicode.html#script. Not sure if the ones in ruby are the subset we support or something else...